parsers/BR.py

```diff
@@ -1,191 +1,179 @@
-    entriesToRemove = (u'hidraulica', u'itaipu50HzBrasil', u'itaipu60Hz', u'total')
-    for k in entriesToRemove:
-        totals.pop(k, None)
-
-    mapped_totals = {generation_mapping.get(name, 'unknown'): val for name, val
-                     in totals.items()}
+    totals["hydro"] = (
+        totals.get("hidraulica", 0.0)
+        + totals.get("itaipu50HzBrasil", 0.0)
+        + totals.get("itaipu60Hz", 0.0)
+    )
+    entries_to_remove = {"hidraulica", "itaipu50HzBrasil", "itaipu60Hz", "total"}
+    mapped_totals = {
+        GENERATION_MAPPING.get(name, "unknown"): val
+        for name, val in totals.items()
+        if name not in entries_to_remove
+    }
 
     return dt, mapped_totals
 
 
 def fetch_production(zone_key, session=None, target_datetime=None, logger=None) -> dict:
     """Requests the last known production mix (in MW) of a given country."""
     if target_datetime:
-        raise NotImplementedError('This parser is not yet able to parse past dates')
+        raise NotImplementedError("This parser is not yet able to parse past dates")
 
     gd = get_data(session, logger)
     generation = production_processor(gd, zone_key)
 
     datapoint = {
-      'zoneKey': zone_key,
-      'datetime': generation[0].datetime,
-      'production': generation[1],
-      'storage': {
-          'hydro': None,
-      },
-      'source': 'ons.org.br'
+        "zoneKey": zone_key,
+        "datetime": generation[0].datetime,
+        "production": generation[1],
+        "storage": {
+            "hydro": None,
+        },
+        "source": SOURCE,
     }
 
-    datapoint = validate(datapoint, logger,
-                         remove_negative=True, required=['hydro'], floor=1000)
+    datapoint = validate(
+        datapoint, logger, remove_negative=True, required=["hydro"], floor=1000
+    )
 
     return datapoint
 
 
-def fetch_exchange(zone_key1, zone_key2, session=None, target_datetime=None, logger=None) -> dict:
+def fetch_exchange(
+    zone_key1, zone_key2, session=None, target_datetime=None, logger=None
+) -> dict:
     """Requests the last known power exchange (in MW) between two regions."""
     if target_datetime:
-        raise NotImplementedError('This parser is not yet able to parse past dates')
+        raise NotImplementedError("This parser is not yet able to parse past dates")
 
     gd = get_data(session, logger)
+    dt = arrow.get(gd["Data"]).datetime
+    scc = "->".join(sorted([zone_key1, zone_key2]))
 
-    if zone_key1 in countries_exchange.keys():
-        country_exchange = countries_exchange[zone_key1]
+    if zone_key1 in COUNTRIES_EXCHANGE.keys():
+        country_exchange = COUNTRIES_EXCHANGE[zone_key1]
 
-    if zone_key2 in countries_exchange.keys():
-        country_exchange = countries_exchange[zone_key2]
+    if zone_key2 in COUNTRIES_EXCHANGE.keys():
+        country_exchange = COUNTRIES_EXCHANGE[zone_key2]
```

Comment:
`keys()` is not needed here because `in <dict>` automatically checks the keys.

In fact this whole thing could be written more concisely as:

```python
    country_exchange = COUNTRIES_EXCHANGE.get(zone_key1) or COUNTRIES_EXCHANGE.get(zone_key2)
```